---
permalink: /
title: "Ben Cao academic personal websites"
excerpt: "About me"
author_profile: true
redirect_from: 
  - /about/
  - /about.html
---
Dr. Cao Ben received his Ph.D. in Computer Science from Dalian University of Technology (DLUT), where he was jointly trained with A*STAR, Singapore. His research focuses on the intersection of computer science and biotechnology, with a commitment to solving biological problems using computational methods. His work spans areas such as DNA data storage and security, coding theory, gene expression, artificial intelligence, and large language models (LLM). Dr. Cao has published over 10 papers as the first author or co-first author in international journals such as Cell Reports, IEEE Transactions, and BIB, including one ESI hot paper and two ESI highly cited papers. He actively contributes to the academic community, serving as the Associate Editor of the Journal of Artificial Intelligence in Bioinformatics, and has been invited to review for approximately 20 SCI journals, including IEEE Transactions on Nanobioscience, IEEE Transactions on Molecular Biological and Multi-Scale Communications, Computational and Structural Biotechnology Journal, Future Generation Computer Systems, Scientific Reports, IEEE Communications Letters, PeerJ, BMC Bioinformatics, PeerJ Computer Science, and others. He has received the National Scholarship twice and the China Scholarship Council (CSC) Scholarship once.

PUBLISHED/ACCEPT (# for co-first author, †for assist supervised master): 
======
1. 	Liu Z#,†, Cao B#, et al. Family of Mutually Uncorrelated Codes for DNA Storage Address Design, IEEE Transactions on NanoBioscience, 2025. DOI: 10.1109/TNB.2025.3530470. 
1. 	Cao B#, Zheng Y#, et al.  Efficient Data Reconstruction: The Bottleneck of Large-scale Application of DNA Storage, Cell reports, 2024. (Cell press, JCR Q1, IF=8.8)
1. 	Cao B#, Wang K#,†, et al.  PELMI: Realize robust DNA image storage under general errors via parity encoding and local mean iteration, Brief in Bioinformatics, 2024. Doi:10.1093/bib/bbae463 (JCR Q1, TOP, IF=6.8)
1. 	Cao B, He T, Li X, et al. Learning Structurally Stabilized Representations for Multi-modal Lossless DNA Storage. arXiv preprint arXiv:2408.00779, 2024.
1. 	Cao B#, Li X#, Wang B, et al. Achieve Handle Level Random Access in Encrypted DNA Archival Storage System via Frequency Dictionary Mapping Coding[J]. bioRxiv, 2024: 2024.08. 15.608111. Major Revision in PATTERNS.
1. 	Li X#, Cao B#, Wang J, et al. Predicting mutation-disease associations through protein interactions via deep learning[J]. bioRxiv, 2024: 2024.08. 06.606730.
1. 	Wang K#,†, Cao B#, Zhao Y, Wang B. DNA-base128: Stores images into DNA via base128 encoding, Journal of Chemical Information and Modeling, 2024. (JCR Q1, TOP, IF=5.6) 
1. 	Cao B, Wang B, Zhang Q.  GCNSA: DNA storage encoding with a graph convolutional network and self-attention, iScience, 2023: 106231. (Cell press, JCR Q1, IF=6.107) 
1. 	Zheng Y#,†, Cao B#, Wu J, et al. High Net Information Density DNA Data Storage by the MOPE Encoding Algorithm, IEEE/ACM Transactions on Computational Biology and Bioinformatics,2023,1-1. (JCR Q1, CCF B, IF=3.702)
1. 	Wang P#,†, Cao B#, Ma T, et al. DUHI: Dynamically updated hash index clustering method for DNA storage. Computers in Biology and Medicine, 2023: 107244. (JCR Q1, IF= 6.698)
1. 	Zhao Y#,†, Cao B#, Wang P, et al. DBTRG: De Bruijn Trim rotation graph encoding for reliable DNA storage. Computational and Structural Biotechnology Journal, 2023, 21: 4469-4477. (JCR Q1, IF= 6.0)
1. 	Mu Z#,†, Cao B#, et al. RBS: A Rotational Coding Based on Blocking Strategy for DNA Storage, IEEE Transactions on NanoBioscience, 2023, 1-1.
1. 	Zhang X, Liu Y, Wang B, Zhou S, Shi P, Cao B, ... & Kirilov Kasabov, N. Biomolecule-Driven Two-Factor Authentication Strategy for Access Control of Molecular Devices. ACS nano, 2023, 17(18), 18178-18189.
1. 	Liu Y, Wang J, Sun L, Wang B, Zhang Q, Zhang X, Cao B. Active Self-Assembly of Ladder-Shaped DNA Carrier for Drug Delivery. Molecules, 2023, 28(2): 797.
1. 	Cao B, Zhang X, Cui S, et al. Adaptive coding for DNA storage with high storage density and low coverage. Npj Systems Biology and Applications,2022,1(8):23. (Nature press, JCR Q1, TOP, IF=4.671)
1. 	Cao B, Shi P, Zheng Y, et al. FMG: An observable DNA storage coding method based on frequency matrix game graphs, Computers in biology and medicine, 2022, pp. 106269. (JCR Q1, IF= 6.698)
1. 	Sun L#,†, Cao B#, et al. TripDesign: A DNA triplex design approach based on interaction forces, Journal of Physical Chemistry B, 2022, 126(43): 8708-8719.
1. 	Cao B, Li X, Zhang X, et al. Designing Uncorrelated Address Constrain for DNA Storage by DMVO Algorithm. IEEE/ACM Transactions on Computational Biology and Bioinformatics,2022,2(19):866-877. (JCR Q1, ESI Top Paper, CCF B, IF=3.702)
1. 	Cao B, Zhang X, Wu J, et al. Minimum free energy coding for DNA storage. IEEE Transactions on NanoBioscience,2021,2:212-222. (ESI Highly Cited Paper, IF=3.206)
1. 	Cao B, Zhao S, Li X, et al. K-Means Multi-Verse Optimizer (KMVO) Algorithm to Construct DNA Storage Codes. IEEE Access,2020,8:29547-29556.
1. 	Yin Q, Cao B, Li X, et al. An Intelligent Optimization Algorithm for Constructing a DNA Storage Code: NOL-HHO. International Journal of Molecular Sciences,2020,6(21). (JCR Q1, TOP, IF=6.208)


A data-driven personal website
======
Like many other Jekyll-based GitHub Pages templates, academicpages makes you separate the website's content from its form. The content & metadata of your website are in structured markdown files, while various other files constitute the theme, specifying how to transform that content & metadata into HTML pages. You keep these various markdown (.md), YAML (.yml), HTML, and CSS files in a public GitHub repository. Each time you commit and push an update to the repository, the [GitHub pages](https://pages.github.com/) service creates static HTML pages based on these files, which are hosted on GitHub's servers free of charge.

Many of the features of dynamic content management systems (like Wordpress) can be achieved in this fashion, using a fraction of the computational resources and with far less vulnerability to hacking and DDoSing. You can also modify the theme to your heart's content without touching the content of your site. If you get to a point where you've broken something in Jekyll/HTML/CSS beyond repair, your markdown files describing your talks, publications, etc. are safe. You can rollback the changes or even delete the repository and start over -- just be sure to save the markdown files! Finally, you can also write scripts that process the structured data on the site, such as [this one](https://github.com/academicpages/academicpages.github.io/blob/master/talkmap.ipynb) that analyzes metadata in pages about talks to display [a map of every location you've given a talk](https://academicpages.github.io/talkmap.html).

Getting started
======
cRegister a GitHub account if you don't have one and confirm your e-mail (required!)
1. Fork [this repository](https://github.com/academicpages/academicpages.github.io) by clicking the "fork" button in the top right. 
1. Go to the repository's settings (rightmost item in the tabs that start with "Code", should be below "Unwatch"). Rename the repository "[your GitHub username].github.io", which will also be your website's URL.
1. Set site-wide configuration and create content & metadata (see below -- also see [this set of diffs](http://archive.is/3TPas) showing what files were changed to set up [an example site](https://getorg-testacct.github.io) for a user with the username "getorg-testacct")
1. Upload any files (like PDFs, .zip files, etc.) to the files/ directory. They will appear at https://[your GitHub username].github.io/files/example.pdf.  
1. Check status by going to the repository settings, in the "GitHub pages" section

Site-wide configuration
------
The main configuration file for the site is in the base directory in [_config.yml](https://github.com/academicpages/academicpages.github.io/blob/master/_config.yml), which defines the content in the sidebars and other site-wide features. You will need to replace the default variables with ones about yourself and your site's github repository. The configuration file for the top menu is in [_data/navigation.yml](https://github.com/academicpages/academicpages.github.io/blob/master/_data/navigation.yml). For example, if you don't have a portfolio or blog posts, you can remove those items from that navigation.yml file to remove them from the header. 

Create content & metadata
------
For site content, there is one markdown file for each type of content, which are stored in directories like _publications, _talks, _posts, _teaching, or _pages. For example, each talk is a markdown file in the [_talks directory](https://github.com/academicpages/academicpages.github.io/tree/master/_talks). At the top of each markdown file is structured data in YAML about the talk, which the theme will parse to do lots of cool stuff. The same structured data about a talk is used to generate the list of talks on the [Talks page](https://academicpages.github.io/talks), each [individual page](https://academicpages.github.io/talks/2012-03-01-talk-1) for specific talks, the talks section for the [CV page](https://academicpages.github.io/cv), and the [map of places you've given a talk](https://academicpages.github.io/talkmap.html) (if you run this [python file](https://github.com/academicpages/academicpages.github.io/blob/master/talkmap.py) or [Jupyter notebook](https://github.com/academicpages/academicpages.github.io/blob/master/talkmap.ipynb), which creates the HTML for the map based on the contents of the _talks directory).

**Markdown generator**

I have also created [a set of Jupyter notebooks](https://github.com/academicpages/academicpages.github.io/tree/master/markdown_generator
) that converts a CSV containing structured data about talks or presentations into individual markdown files that will be properly formatted for the academicpages template. The sample CSVs in that directory are the ones I used to create my own personal website at stuartgeiger.com. My usual workflow is that I keep a spreadsheet of my publications and talks, then run the code in these notebooks to generate the markdown files, then commit and push them to the GitHub repository.

How to edit your site's GitHub repository
------
Many people use a git client to create files on their local computer and then push them to GitHub's servers. If you are not familiar with git, you can directly edit these configuration and markdown files directly in the github.com interface. Navigate to a file (like [this one](https://github.com/academicpages/academicpages.github.io/blob/master/_talks/2012-03-01-talk-1.md) and click the pencil icon in the top right of the content preview (to the right of the "Raw | Blame | History" buttons). You can delete a file by clicking the trashcan icon to the right of the pencil icon. You can also create new files or upload files by navigating to a directory and clicking the "Create new file" or "Upload files" buttons. 

Example: editing a markdown file for a talk
![Editing a markdown file for a talk](/images/editing-talk.png)

For more info
------
More info about configuring academicpages can be found in [the guide](https://academicpages.github.io/markdown/). The [guides for the Minimal Mistakes theme](https://mmistakes.github.io/minimal-mistakes/docs/configuration/) (which this theme was forked from) might also be helpful.
